### ISSUE TITLE
Parameters for fine-tuning models with CPU backend? 

### ISSUE THREAD

**Issue by gmurthy** [facebookresearch/flashlight#510]
I'm exploring fine-tuning of the pre-trained models. To start with, I'm just trying to run the fine-tuning tutorial using the provided AMI corpus, but with a CPU backend. However, I'm unable to get training to go past the first iteration of epoch 1. As soon as it starts, training aborts with the following error: **FinetuneCTC.cpp:613] Loss has NaN values.**

```
I0330 17:32:15.987434  9086 FinetuneCTC.cpp:276] [Network Params: 70498735]
I0330 17:32:15.987478  9086 FinetuneCTC.cpp:281] [Criterion] ConnectionistTemporalClassificationCriterion
I0330 17:32:16.038491  9086 FinetuneCTC.cpp:285] [Network Optimizer] SGD (momentum=0.8)
I0330 17:32:16.041868  9086 FinetuneCTC.cpp:538] Shuffling trainset
I0330 17:32:16.042310  9086 FinetuneCTC.cpp:545] Epoch 1 started!
F0330 17:32:24.966100  9086 FinetuneCTC.cpp:613] Loss has NaN values. Samples - ES2008d_H00_FEE029_2232.12_2233.38,ES2015d_H03_FEE060_1196.33_1197.58,ES2005c_H03_FEE019_2233.32_2234.57,TS3009b_H01_MTD035UID_1164.82_1166.04
*** Check failure stack trace: ***
    @     0x7fef22a6e0cd  google::LogMessage::Fail()
    @     0x7fef22a6ff33  google::LogMessage::SendToLog()
    @     0x7fef22a6dc28  google::LogMessage::Flush()
    @     0x7fef22a70999  google::LogMessageFatal::~LogMessageFatal()
    @     0x5616c6055b13  (unknown)
    @     0x7fef124fabf7  __libc_start_main
    @     0x5616c60d77ca  (unknown)
[ff84fdd979b8:09089] *** Process received signal ***
[ff84fdd979b8:09089] Signal: Segmentation fault (11)
[ff84fdd979b8:09089] Signal code: Address not mapped (1)
[ff84fdd979b8:09089] Failing at address: 0x7fe0c7a7920d
[ff84fdd979b8:09089] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12980)[0x7fe0ca520980]
[ff84fdd979b8:09089] [ 1] /lib/x86_64-linux-gnu/libc.so.6(getenv+0xa5)[0x7fe0ca15f8a5]
[ff84fdd979b8:09089] [ 2] /usr/lib/x86_64-linux-gnu/libtcmalloc.so.4(_ZN13TCMallocGuardD1Ev+0x34)[0x7fe0ca9cae44]
[ff84fdd979b8:09089] [ 3] /lib/x86_64-linux-gnu/libc.so.6(__cxa_finalize+0xf5)[0x7fe0ca160735]
[ff84fdd979b8:09089] [ 4] /usr/lib/x86_64-linux-gnu/libtcmalloc.so.4(+0x13cb3)[0x7fe0ca9c8cb3]
[ff84fdd979b8:09089] *** End of error message ***
```

The behavior is identical in the Colab, as well as on my computer (Ubuntu 16.04 x86-64). It also remains the same irrespective of whether the I use the provided dataset (AMI corpus) or try my own.

It seems like the training parameters in the tutorial are optimized for the GPU back-end, but are there any pointers for how to run fine-tuning with the CPU back-end? I assume it is supported and expected to work on the CPU. Any help you could provide would be very useful.

**Comment by jacobkahn** [comment]
@gmurthy — if you built Flashlight from source, can you run `make test` in your Flashlight build directory and ensure all of your tests are passing with the CPU backend?

**Comment by abhinavdayal** [comment]
I am also getting the error, while running using CPU backend. 

### Error

fl_asr_tutorial_finetune_ctc model.bin --datadir ami_limited_supervision --train train_10min_0.lst,train_10min_1.lst,train_10min_2.lst,train_10min_3.lst,train_10min_4.lst,train_10min_5.lst,train_9hr.lst --valid dev:dev.lst --arch arch.txt --tokens tokens.txt --lexicon lexicon.txt --rundir checkpoint --lr 0.025 --netoptim sgd --momentum 0.8 --reportiters 1000 --lr_decay 100 --lr_decay_step 50 --iter 25000 --batchsize 4 --warmup 0
Initialized Gloo successfully!
F20211229 17:59:02.459493 19752 FinetuneCTC.cpp:615] Loss has NaN values. Samples - ES2014b_H02_FEE055_977.04_980.75,ES2013c_H00_FEE049_2200.73_2204.44,ES2015b_H02_FEE059_583.73_587.44,ES2008d_H01_FEE030_1631.49_1635.2
*** Check failure stack trace: ***
    @     0x7f03f48b8e0a  google::LogMessage::Fail()
    @     0x7f03f48b8d4c  google::LogMessage::SendToLog()
    @     0x7f03f48b85dc  google::LogMessage::Flush()
    @     0x7f03f48bc276  google::LogMessageFatal::~LogMessageFatal()
    @     0x55f23f8067c8  _ZZ4mainENKUlSt10shared_ptrIN2fl6ModuleEES_INS0_3app3asr17SequenceCriterionEES_INS0_7DatasetEES_INS0_19FirstOrderOptimizerEEdlE3_clES2_S6_S8_SA_dl
    @     0x55f23f7688a5  main
    @     0x7f03e8c3d0b3  __libc_start_main
    @     0x55f23f7ff0ee  _start
    @              (nil)  (unknown)
Aborted

### Inference/tests are working
fl_asr_test --am model.bin --datadir '' --emission_dir '' --uselexicon false \
            --test ami_limited_supervision/test.lst --tokens tokens.txt --lexicon lexicon.txt --show

works fine.

### failing make tests
Out of 409 only the following three tests do not pass in make test (Tried  various iterations of installing the dependencies right to make all tests pass):

The following tests FAILED:
        357 - Seq2SeqTest.Seq2SeqBeamSearchViterbi (Failed)
        358 - Seq2SeqTest.Seq2SeqMedianWindow (Failed)
        359 - Seq2SeqTest.Seq2SeqStepWindow (Failed)

**Comment by davidbelle** [comment]
Getting the same symptoms on CPU only. Any updates or ideas on this?
